hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/timeline/HoodieTimelineArchiver.java

```diff
@@ -217,6 +217,10 @@ private List<HoodieInstant> getCommitInstantsToArchive() throws IOException {
           earliestInstantToRetainCandidates.add(
               completedCommitsTimeline.findInstantsModifiedAfterByCompletionTime(latestCompactionTime.get()).firstInstant());
         }
+      } catch (UnsupportedOperationException unsupportedOperationException) {
+        // If tableMetadata is FileSystemBackedTableMetadata would throw UnsupportedOperationException, should skip it to
+        // confirm next operation success
+        LOG.warn("tableMetadata is FileSystemBackedTableMetadata and skip limiting archiving of instants.");
```

Comment:
Why the metadata backend is fs based if metadata is available?

Reply:
![1715846706392.png](https://github.com/apache/hudi/assets/10645422/9fcc7240-3975-4397-8106-d4195871f384)

Reply:
The `isMetadataTableAvailable` checks the `hoodie.properties` for `FILES` metadata partition instead, it should be empty if the MDT does not initialize successfully.

Reply:
The condition is mdt with enable firstly , and make it disable after several insert. Then mdt table run archive would ocurr  this sences. At this condition mdt dir is not empty and would occur the error. @danny0405

Reply:
> The condition is mdt with enable firstly , and make it disable after several insert

Then the `isMetadataTableAvailable` should be false? Beause once the mdt is disabled, the `FILES` partition would also be removed from `hoodie.properties`.

Reply:
Emm，but the logic stack is inexplicable，do you think the guard check for UnsupportedOperationException is neccessary?  @danny0405

Reply:
Not really because it is unexpected. I guess there might be some inconsistency for metadata table configurations.